manytask/models.py

```
@@ -62,13 +62,36 @@ class UserOnCourse(Base):
     # relationships
     user: Mapped['User'] = relationship(back_populates='users_on_courses')
     course: Mapped['Course'] = relationship(back_populates='users_on_courses')
+    grades: DynamicMapped['Grade'] = relationship(
+        back_populates='user_on_course')
+
+
+@event.listens_for(UserOnCourse, 'before_insert')
+def validate_gitlab_instance_host(mapper: Mapper[UserOnCourse], connection: Connection, target: UserOnCourse) -> None:
+    session = Session(bind=connection)
+
+    try:
+        if target.user:
+            user = target.user
+        else:
+            user = session.query(User).filter_by(id=target.user_id).one()
+
+        if target.course:
+            course = target.course
+        else:
+            course = session.query(Course).filter_by(id=target.course_id).one()
+    except Exception:
+        return
```

Why do you need to swallow exception here?

You are catching exception and after that just return, this is antipattern, do we have a good reason for doing this?

---

Because if there is a problem with finding user or course, we don't want to think about exceptions here. We swallow the exception so that sqlalchemy.orm raises the necessary exception for us when trying to create an object itself.

---

@gagarinkomar I think the worst part of this is catching a The Most General Exception Posssible (TM). Maybe NoResultFound would be better?

---

@gagarinkomar can you add logging here and a todo?
Seems like a funny place to swallow errors later.

---

https://github.com/manytask/manytask/issues/219 added issue